src/Datadog.Trace/Span.cs

```diff
@@ -295,7 +278,11 @@ internal Span SetMetric(string key, int? value)
 
         internal void LockSamplingPriority()
         {
-            _samplingPriorityLocked = true;
+            // only set this metric on root spans
+            if (Context.ParentId > 0 && Context.SamplingPriority != null)
```

Comment:
I think we want this logic to be: "If this span is the local root span, and we have a sampling priority, set the tag"

Meaning, we still want to set this tag even if we inherited a parent id from a distributed trace, but only on the first span of the trace for this service.

This could just be `if (Parent == null && Context.SamplingPriority != null)` ? (I am guessing)